packages/features/form-builder/FormBuilderField.tsx

```diff
@@ -0,0 +1,343 @@
+      {/* multiemail doesnt show label initially. It is shown on clicking CTA */}
+      {/* boolean type doesn't have a label overall, the radio has it's own label */}
+      {/* Component itself managing it's label should remove these checks */}
+      {field.type !== "boolean" && field.type !== "multiemail" && field.label && (
+        <div className="mb-2 flex items-center">
+          <Label className="!mb-0">
+            <span>{field.label}</span>
+            <span className="text-emphasis ml-1 -mb-1 text-sm font-medium leading-none">
+              {!readOnly && field.required ? "*" : ""}
+            </span>
+          </Label>
+        </div>
+      )}
+      {children}
+    </div>
+  );
+};
+
+/**
+ * Ensures that `labels` and `placeholders`, wherever they are, are set properly. If direct values are not set, default values from fieldTypeConfig are used.
+ */
+function getAndUpdateNormalizedValues(field: RhfFormFields[number], t: TFunction) {
+  let noLabel = false;
+  let hidden = !!field.hidden;
+  if (field.type === "radioInput") {
+    const options = field.options;
+
+    // If we have only one option and it has an input, we don't show the field label because Option name acts as label.
+    // e.g. If it's just Attendee Phone Number option then we don't show `Location` label
+    if (options?.length === 1) {
+      if (!field.optionsInputs) {
+        throw new Error("radioInput must have optionsInputs");
+      }
+      if (field.optionsInputs[options[0].value]) {
+        noLabel = true;
+      } else {
+        // If there's only one option and it doesn't have an input, we don't show the field at all because it's visible in the left side bar
+        hidden = true;
+      }
+    }
+  }
+
+  const label = noLabel ? "" : field.label || t(field.defaultLabel || "");
+  const placeholder = field.placeholder || t(field.defaultPlaceholder || "");
+
+  if (field.variantsConfig?.variants) {
+    Object.entries(field.variantsConfig.variants).forEach(([variantName, variant]) => {
+      variant.fields.forEach((variantField) => {
+        const fieldTypeVariantsConfig = fieldTypesConfigMap[field.type]?.variantsConfig;
+        const defaultVariantFieldLabel =
+          fieldTypeVariantsConfig?.variants?.[variantName]?.fieldsMap[variantField.name]?.defaultLabel;
+
+        variantField.label = variantField.label || t(defaultVariantFieldLabel || "");
+      });
+    });
+  }
+
+  return { hidden, placeholder, label };
+}
+
+export const ComponentForField = ({
+  field,
+  value,
+  setValue,
+  readOnly,
+}: {
+  field: Omit<RhfFormField, "editable" | "label"> & {
+    // Label is optional because radioInput doesn't have a label
+    label?: string;
+  };
+  readOnly: boolean;
+} & ValueProps) => {
+  const fieldType = field.type || "text";
+  const componentConfig = Components[fieldType];
+
+  const isValueOfPropsType = (val: unknown, propsType: typeof componentConfig.propsType) => {
+    const isValid = isValidValueProp[propsType](val);
```

Comment:
`isValidValueProp` is now in Components.tsx as it is more closely related with the Component itself.